google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/StreamingSubscriberConnection.java

```diff
@@ -95,48 +105,79 @@
    */
   private final String clientId = UUID.randomUUID().toString();
 
-  public StreamingSubscriberConnection(
-      String subscription,
-      MessageReceiver receiver,
-      Duration ackExpirationPadding,
-      Duration maxAckExtensionPeriod,
-      Duration maxDurationPerAckExtension,
-      Distribution ackLatencyDistribution,
-      SubscriberStub stub,
-      int channelAffinity,
-      FlowControlSettings flowControlSettings,
-      boolean useLegacyFlowControl,
-      FlowController flowController,
-      ScheduledExecutorService executor,
-      ScheduledExecutorService systemExecutor,
-      ApiClock clock) {
-    this.subscription = subscription;
-    this.systemExecutor = systemExecutor;
-    if (maxDurationPerAckExtension.compareTo(DEFAULT_MAX_DURATION_PER_ACK_EXTENSION) == 0) {
-      this.streamAckDeadline = DEFAULT_STREAM_ACK_DEADLINE;
-    } else if (maxDurationPerAckExtension.compareTo(MIN_STREAM_ACK_DEADLINE) < 0) {
-      this.streamAckDeadline = MIN_STREAM_ACK_DEADLINE;
-    } else if (maxDurationPerAckExtension.compareTo(MAX_STREAM_ACK_DEADLINE) > 0) {
-      this.streamAckDeadline = MAX_STREAM_ACK_DEADLINE;
+  private StreamingSubscriberConnection(Builder builder) {
+    subscription = builder.subscription;
+    systemExecutor = builder.systemExecutor;
+
+    // We need to set the default stream ack deadline on the initial request, this will be
+    // updated by modack requests in the message dispatcher
+    if (builder.maxDurationPerAckExtensionDefaultUsed) {
+      // If the default is used, check if exactly once is enabled and set appropriately
+      if (builder.exactlyOnceDeliveryEnabled) {
+        streamAckDeadlineSeconds.set(
+            Math.toIntExact(Subscriber.STREAM_ACK_DEADLINE_EXACTLY_ONCE_DEFAULT.getSeconds()));
```

Comment:
exactly_once_delivery* here and elsewhere.

Reply:
Updated.